pkg/daemonset/daemonset.go

```diff
@@ -138,6 +138,51 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies HostNetwork to daemonset definition.
+func (builder *Builder) WithHostNetwork() *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Enabling hostnetwork flag to daemonset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.HostNetwork = true
+
+	return builder
+}
+
+// WithConfigMapVolume defines Volume of daemonset under PodTemplateSpec with given ConfigMap.
+func (builder *Builder) WithConfigMapVolume(cmName string) *Builder {
```

Comment:
Hey sorry for the confusion but the idea is to just make this a generic WithVolume function. We don't want to limit it to only configmaps and creating a new mutation for each volume source type would be a lot of work and difficult to manage. For the most part it is just a copy-paste of with WithVolume method from the deployment pkg